### PR TITLE
Fix Sonar issues

### PR DESCRIPTION
**Pull request by robbertnoordzij**
---

### Description

---

I was not sure if someone was already fixing the NPE warning from Sonar. I also removed the code duplication that was introduced with my previous PR and I added the generic hints to the NameNode.

Changes were made to:
- [x] Codegen library - Java
- [x] Codegen library - Kotlin
- [x] Codegen library - Scala
- [ ] Maven plugin
- [ ] Gradle plugin
- [ ] SBT plugin

### CODE REVIEW

File: src/main/java/com/kobylynskyi/graphql/codegen/mapper/GraphQLTypeMapper.java [hunk 1]
```diff
@@ -270,7 +271,11 @@ default NamedDefinition getLanguageType(MappingContext mappingContext, String gr
     }
 
     default List<String> getAnnotations(MappingContext mappingContext, ExtendedDefinition<?, ?> extendedDefinition) {
-        NamedNode def = extendedDefinition != null ? extendedDefinition.getDefinition() : null;
+        if(Objects.isNull(extendedDefinition)) {
+            return Collections.emptyList();
+        }
```
Comment (kobylynskyi) [inline src/main/java/com/kobylynskyi/graphql/codegen/mapper/GraphQLTypeMapper.java:276]:
```suggestion
        if (extendedDefinition == null) {
            return Collections.emptyList();
        }
```

File: src/main/java/com/kobylynskyi/graphql/codegen/model/definitions/ExtendedDefinition.java [hunk 2]
```diff
@@ -14,6 +14,9 @@
 import java.util.Collection;
 import java.util.List;
 import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static java.util.function.Predicate.not;
```
Comment (kobylynskyi) [inline src/main/java/com/kobylynskyi/graphql/codegen/model/definitions/ExtendedDefinition.java:19]: Predicate#not was released only in Java 11: https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/function/Predicate.html#not(java.util.function.Predicate)
Could you please make a backwards-compatible change that is supported by Java 8 as well?
Thanks.

Comment (kobylynskyi) [inline src/main/java/com/kobylynskyi/graphql/codegen/model/definitions/ExtendedDefinition.java:19]:
```suggestion
```

File: src/main/java/com/kobylynskyi/graphql/codegen/model/definitions/ExtendedDefinition.java [hunk 3]
```diff
@@ -127,10 +121,12 @@ public SourceLocation getSourceLocation() {
...
                 directives.addAll(definitionDirectives);
             }
-            this.extensions.stream().filter(Objects::nonNull)
+            this.extensions.stream()
+                .filter(Objects::nonNull)
                 .map(DirectivesContainer.class::cast)
-                .map(DirectivesContainer::getDirectives).filter(Objects::nonNull)
-                .forEach(ds -> ds.forEach(d -> directives.add(((Directive) d))));
+                .map(DirectivesContainer::getDirectives)
+                .filter(not(Utils::isEmpty))
```
Comment (kobylynskyi) [inline src/main/java/com/kobylynskyi/graphql/codegen/model/definitions/ExtendedDefinition.java:128]:
```suggestion
                .filter(c -> !Utils.isEmpty(c))
```

**Review by kobylynskyi — CHANGES_REQUESTED** [review]

File: src/main/java/com/kobylynskyi/graphql/codegen/model/definitions/ExtendedDefinition.java (hunk 2 above)
Comment (robbertnoordzij) [reply src/main/java/com/kobylynskyi/graphql/codegen/model/definitions/ExtendedDefinition.java]: I took your suggestion and applied it!

**Review by jxnu-liguobin — APPROVED** [review]

**Review by kobylynskyi — APPROVED** [review]

**Pull request merged**